### ISSUE TITLE
[REVIEW]: From Maps to Models - Tutorials for structural geological modeling using GemPy and GemGIS

### ISSUE THREAD

**Issue by whedon** [openjournals/jose-reviews#185]
**Submitting author:** @AlexanderJuestel (<a href="http://orcid.org/0000-0003-0980-7479">Alexander Jüstel</a>)
**Repository:** <a href="https://github.com/cgre-aachen/gemgis_data" target ="_blank">https://github.com/cgre-aachen/gemgis_data</a>
**Version:** v.1.0.0
**Editor:** @acocac
**Reviewer:** @BenjMy, @jwagemann
**Archive:** Pending

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://jose.theoj.org/papers/5e6ad1eedf43a1175c41e188b2881941/status.svg)](https://jose.theoj.org/papers/5e6ad1eedf43a1175c41e188b2881941)

Status badge code:

```
HTML: <a href="https://jose.theoj.org/papers/5e6ad1eedf43a1175c41e188b2881941"><img src="https://jose.theoj.org/papers/5e6ad1eedf43a1175c41e188b2881941/status.svg"></a>
Markdown: [![status](https://jose.theoj.org/papers/5e6ad1eedf43a1175c41e188b2881941/status.svg)](https://jose.theoj.org/papers/5e6ad1eedf43a1175c41e188b2881941)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/cgre-aachen/gemgis_data" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@BenjMy & @jwagemann, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/jose-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @acocac know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [ ] As the reviewer I confirm that I have read the [JOSE conflict of interest policy](https://github.com/openjournals/joss/blob/jose/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSE code of conduct](https://jose.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source for this learning module available at the <a target="_blank" href="https://github.com/cgre-aachen/gemgis_data">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of a standard license? (OSI-approved for code, Creative Commons for content)
- [ ] **Version:** Does the release version given match the repository release (v.1.0.0)?
- [ ] **Authorship:** Has the submitting author (@AlexanderJuestel) made visible contributions to the module? Does the full list of authors seem appropriate and complete?

### Documentation

- [ ] **A statement of need:** Do the authors clearly state the need for this module and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly stated list of dependencies?
- [ ] **Usage:** Does the documentation explain how someone would adopt the module, and include examples of how to use it?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the module 2) Report issues or problems with the module 3) Seek support

### Pedagogy / Instructional design (Work-in-progress: reviewers, please comment!)

- [ ] **Learning objectives:** Does the module make the learning objectives plainly clear? (We don't require explicitly written learning objectives; only that they be evident from content and design.)
- [ ] **Content scope and length:** Is the content substantial for learning a given topic? Is the length of the module appropriate?
- [ ] **Pedagogy:** Does the module seem easy to follow? Does it observe guidance on cognitive load? (working memory limits of 7 +/- 2 chunks of information)
- [ ] **Content quality:** Is the writing of good quality, concise, engaging? Are the code components well crafted? Does the module seem complete?
- [ ] **Instructional design:** Is the instructional design deliberate and apparent? For example, exploit worked-example effects; effective multi-media use; low extraneous cognitive load.

### JOSE paper

- [ ] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Does the paper clearly state the need for this module and who the target audience is?
- [ ] **Description:** Does the paper describe the learning materials and sequence?
- [ ] Does it describe how it has been used in the classroom or other settings, and how someone might adopt it?
- [ ] Could someone else teach with this module, given the right expertise?
- [ ] Does the paper tell the "story" of how the authors came to develop it, or what their expertise is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [ ] As the reviewer I confirm that I have read the [JOSE conflict of interest policy](https://github.com/openjournals/joss/blob/jose/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSE code of conduct](https://jose.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source for this learning module available at the <a target="_blank" href="https://github.com/cgre-aachen/gemgis_data">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of a standard license? (OSI-approved for code, Creative Commons for content)
- [ ] **Version:** Does the release version given match the repository release (v.1.0.0)?
- [ ] **Authorship:** Has the submitting author (@AlexanderJuestel) made visible contributions to the module? Does the full list of authors seem appropriate and complete?

### Documentation

- [ ] **A statement of need:** Do the authors clearly state the need for this module and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly stated list of dependencies?
- [ ] **Usage:** Does the documentation explain how someone would adopt the module, and include examples of how to use it?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the module 2) Report issues or problems with the module 3) Seek support

### Pedagogy / Instructional design (Work-in-progress: reviewers, please comment!)

- [ ] **Learning objectives:** Does the module make the learning objectives plainly clear? (We don't require explicitly written learning objectives; only that they be evident from content and design.)
- [ ] **Content scope and length:** Is the content substantial for learning a given topic? Is the length of the module appropriate?
- [ ] **Pedagogy:** Does the module seem easy to follow? Does it observe guidance on cognitive load? (working memory limits of 7 +/- 2 chunks of information)
- [ ] **Content quality:** Is the writing of good quality, concise, engaging? Are the code components well crafted? Does the module seem complete?
- [ ] **Instructional design:** Is the instructional design deliberate and apparent? For example, exploit worked-example effects; effective multi-media use; low extraneous cognitive load.

### JOSE paper

- [ ] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Does the paper clearly state the need for this module and who the target audience is?
- [ ] **Description:** Does the paper describe the learning materials and sequence?
- [ ] Does it describe how it has been used in the classroom or other settings, and how someone might adopt it?
- [ ] Could someone else teach with this module, given the right expertise?
- [ ] Does the paper tell the "story" of how the authors came to develop it, or what their expertise is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @BenjMy, @jwagemann it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/jose-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/jose-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
Wordcount for `paper.md` is 2280

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=2.14 s (31.8 files/s, 52267.7 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Jupyter Notebook                49              0         105485           5289
Markdown                         7            107              0            268
QML                              1              3              0            240
XML                              8              0              0            221
TeX                              1             14              0            168
YAML                             2              5              6             34
-------------------------------------------------------------------------------
SUM:                            68            129         105491           6220
-------------------------------------------------------------------------------


Statistical information for the repository '537c091cba9272b7a32d1dce' was
gathered on 2022/12/01.
No commited files with the specified extensions were found.
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00185/jose.00185/10.21105.jose.00185.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00185/jose.00185/10.21105.jose.00185.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.5281/zenodo.4569086 is OK
- 10.5194/gmd-12-1-2019 is OK
- 10.21105/joss.01450 is OK
- 10.5281/zenodo.4572994 is OK
- 10.1109/MCSE.2007.55 is OK
- 10.1016/j.pepi.2008.06.013 is OK
- 10.1007/978-94-011-2556-7_11 is OK
- 10.1130/GES02455.1 is OK
- 10.1007/BF02775087 is OK
- 10.1007/978-1-4615-9630-1 is OK
- 10.21105/joss.03709 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by acocac** [comment]
👋 @BenjMy & @jwagemann we will conduct the review in this issue.

Please read through the above information and let me know if you have any questions about process.

Thank you 🙏

**Comment by whedon** [comment]
:wave: @BenjMy, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @jwagemann, please update us on how your review is going (this is an automated reminder).

**Comment by BenjMy** [comment]
Thanks for the reminder, I should be able to send my review by the end of next week.

**Comment by BenjMy** [comment]
@AlexanderJuestel @acocac

I have completed an initial review of the GemGIS tutorial. First I'd like to congratulate the GemGIS dev team for this nice piece of work. I took the measure of all the time and investment made to build the tools and notebooks and make them accessible for everyone.

This package has the potential to make an immediate impact on the scientific communities. There are thoughtful and well-paced tutorial notebooks that provide high-level introductions and more detailed examples of the various methods offered in GemGIS (and Gempy). I am excited both for the package's potential in the long run and possibly to use it in the immediate future.

Below are some of my initial thoughts from the checklist that I will begin creating issues for and linking back to this comment.

# Checklist topics
 - **General Checks**
   - Version: up to now there is no release of the proposed package. 
   - Contribution and authorship: unclear what role Miguel de la Varga, Nils Chudalla, and Stefan Back have played in the code development (no commits).

 - **Documentation**

   - Readers are invited to refer to the GEMGIS core code documentation since there is no specific documentation for the GEMGIS-data repos. I think hosting the datasets and notebooks in a separate repo is perfectly relevant particularly to provide clear guidelines on the data **FAIRness**.

     - I suggest adding details (add a section in the documentation) covering how FAIR are the dataset provided:
     - The LICENCE file covers the GemGIS software rights. But under what Licence are the data distributed? I've seen that the authors cited individually the map source for each notebook but it should also clearly be stated on the README and in the doc. 
     - Is there a plan for long-term archiving of the data? into Zenodo for instance?
     - Is there a plan for version-controlled data? data could be fetched (using Pooch for example). 
  

   - _A statement of need_
     - The authors don't identify the target audience in the docs nor in the README. 
     - The definition of the problem being solved is only part of the doc. Would be also interesting to add it to the README.
   - _Installation instructions_ 
     - Issues with the badge target link
     - The is no list of dependencies in the README. At least the authors should make clear that Gempy is an optional dependency.
     - Error using conda-forge installation (inconsistent env) 
     - Error using pip installation (missing dependencies)
     - In the doc, reorder and put "Installing Gempy (Optional)" at the end of the list.

   - _Usage_
     - I opened an issue with some recommendations.    
     - Error Using Binder when GEMGIS is imported.

   - _Community guidelines_:
     - Add "Authorship and credit" statement.
     - I would detail a bit more what can of data can be submitted. Under what terms?

     
   
 - **Pedagogy / Instructional design**
 
   - _Learning objectives_
   
   Individually each notebook is thoroughly documented to guide the user and it is really much appreciated. Nevertheless, the documentation design with respect to the learning objective can be improved. I recommend the authors articulate the notebooks over sections and subsections. Ideally, I would distinguish within new section tutorials which are Gempy dependent from the others. Also, you could articulate the notebooks using sections following your "Learning Objectives" workflow. 
      
   - _Instructional design_
     - Video Tutorials are expected and I think it is a great initiative. 


 - **Jose Paper**
 
Overall it is a good start but it reads a bit choppy. I found the article very resource exploitation (oil, mining, ...) target-oriented while I think this tool has many other disciplines to reach.
 
   - _Statement of needs_
   	
     - To me for instance the tools developed can be also associated with environmental risk management (water scarcity, landslides, ...). I am personally interested in this package for Catchment hydrogeology modelling. Authors should broaden this idea in the statement of needs.
   
     - I like the proof of concept and that the authors already used the software to teach RWTH Aachen. Could it be possible to add (in the "Experience of use in teaching and learning situations" section): 
     	- a link to the degree where it has been taught;
     	- the number of hours dedicated to the course (and how it has been distributed over all the notebooks);
     	- the ratio between the number of instructors and the number of students.
     - "Even from seismic data" is to me again to me narrow. Why other geophysical approaches able to infer geological structures (TEM, gravity, ...) are not considered? 
      
      
   - _Ressources_:    
     - I suggest changing the order of resources and placing "GemGIS" first and then adding "GemPy" indicating that it is an optional dependency. 
     - "Tutorial repository" link and "GemGIS link" are redundant. Choose one or another or change the first to "https://gemgis.readthedocs.io/en/latest/getting_started/tutorial/index.html"

   - _References_:  it looks like the references were broken during the paper compilation.

**Comment by jwagemann** [comment]
@acocac I also concluded my review and added my comments to the issues @BenjMy opened in the target repository.

I'd like to thank the authors for submitting their tutorial to JOSE and I think that the notebooks have a great potential to be a valuable contribution for the GemGIS and GemPy communities. However, at this stage, my evaluation is that the tutorials require a major review as in my opinion pedagogical and educational standards of notebooks are not met.

There is room to improve the structure of the course, but also the quality of the notebooks in terms of including a pedagogical narrative. Once the notebooks have been better structured and improved, the paper submission needs a revision as well. There is also potential to improve structure, clarity and guidance.

My proposals on how to improve the submission are found as a response to the following issues:
* https://github.com/cgre-aachen/gemgis_data/issues/20 
* https://github.com/cgre-aachen/gemgis_data/issues/19
* https://github.com/cgre-aachen/gemgis_data/issues/18

**Comment by AlexanderJuestel** [comment]
Dear @BenjMy, 
Dear @jwagemann, 

thank you very much for your extensive reviews. I will try to resolve all issues during the Christmas break! We certainly have a lot of things to do here! 

Cheers 
Alex

**Comment by acocac** [comment]
Hello, @AlexanderJuestel  👋
@jwagemann @BenjMy have kindly provided their feedback and suggestions to improve the submission. 
Please work on your revisions and come back here to let us know when you want reviewers and myself to take a new look.
Wishing a happy Christmas break.

**Comment by AlexanderJuestel** [comment]
Hey @BenjMy,
I would need some feedback on the issues you opened in the GemGIS Data Repo to continue with my revisions ;) 

Thanks and Cheers
Alex

**Comment by BenjMy** [comment]
Hi @AlexanderJuestel!
I'm on vacation break until January 10th. I'll get back to you asap. 
Cheers,

**Comment by AlexanderJuestel** [comment]
I copied your @BenjMy  list to better track my progress. 

> # Checklist topics
> * **General Checks**
>   
>  

> - [x] Version: up to now there is no release of the proposed package. --> **As agreed in https://github.com/cgre-aachen/gemgis_data/issues/17, there will be no further release of the package as the package does not contain any code but only data**
> - [x] Contribution and authorship: unclear what role Miguel de la Varga, Nils Chudalla, and Stefan Back have played in the code development (no commits). --> **A author statement was added to the JOSE Paper , see https://github.com/cgre-aachen/gemgis_data/commit/32e6fcafb79d17f9801aa5f07730be4f65e3930a**


> * **Documentation**
>   
>   * Readers are invited to refer to the GEMGIS core code documentation since there is no specific documentation for the GEMGIS-data repos. I think hosting the datasets and notebooks in a separate repo is perfectly relevant particularly to provide clear guidelines on the data **FAIRness**.
>     
> - [ ]  I suggest adding details (add a section in the documentation) covering how FAIR are the dataset provided:
> - [ ]  The LICENCE file covers the GemGIS software rights. But under what Licence are the data distributed? I've seen that the authors cited individually the map source for each notebook but it should also clearly be stated on the README and in the doc.
> - [x]  Is there a plan for long-term archiving of the data? into Zenodo for instance? --> **A Zenodo Repository will be created upon acceptance of the paper**
> - [x]  Is there a plan for version-controlled data? data could be fetched (using Pooch for example). --> **All files are already stored on a OwnCloud server and accessible via pooch, the notebooks will be adapted to either use the shipped data in the repository or allow the user to download the data for each notebook**
>   * _A statement of need_
>     
> - [ ]  The authors don't identify the target audience in the docs nor in the README.
> - [ ]  The definition of the problem being solved is only part of the doc. Would be also interesting to add it to the README.
>   * _Installation instructions_
>     
> - [ ] Issues with the badge target link
> - [ ]  The is no list of dependencies in the README. At least the authors should make clear that Gempy is an optional dependency.
> - [ ]  Error using conda-forge installation (inconsistent env)
> - [ ]  Error using pip installation (missing dependencies)
> - [ ]  In the doc, reorder and put "Installing Gempy (Optional)" at the end of the list.
>   * _Usage_
>     
> - [ ]  I opened an issue with some recommendations.
> - [ ]  Error Using Binder when GEMGIS is imported.
>   * _Community guidelines_:
>     
> - [x]  Add "Authorship and credit" statement. -->  **A author statement was added to the JOSE Paper , see https://github.com/cgre-aachen/gemgis_data/commit/32e6fcafb79d17f9801aa5f07730be4f65e3930a**

> - [ ]  I would detail a bit more what can of data can be submitted. Under what terms?
> * **Pedagogy / Instructional design**
>   
>   * _Learning objectives_
>   
>   Individually each notebook is thoroughly documented to guide the user and it is really much appreciated. Nevertheless, the documentation design with respect to the learning objective can be improved. I recommend the authors articulate the notebooks over sections and subsections. Ideally, I would distinguish within new section tutorials which are Gempy dependent from the others. Also, you could articulate the notebooks using sections following your "Learning Objectives" workflow.
>   
>   * _Instructional design_
>     
>     * Video Tutorials are expected and I think it is a great initiative.
> * **Jose Paper**
> 
> Overall it is a good start but it reads a bit choppy. I found the article very resource exploitation (oil, mining, ...) target-oriented while I think this tool has many other disciplines to reach.
> 
> * _Statement of needs_
>   
> - [ ] To me for instance the tools developed can be also associated with environmental risk management (water scarcity, landslides, ...). I am personally interested in this package for Catchment hydrogeology modelling. Authors should broaden this idea in the statement of needs.
> - [ ]  I like the proof of concept and that the authors already used the software to teach RWTH Aachen. Could it be possible to add (in the "Experience of use in teaching and learning situations" section):
>     
> - [ ]  a link to the degree where it has been taught;
> - [ ]  the number of hours dedicated to the course (and how it has been distributed over all the notebooks);
> - [ ]  the ratio between the number of instructors and the number of students.
> - [ ]  "Even from seismic data" is to me again to me narrow. Why other geophysical approaches able to infer geological structures (TEM, gravity, ...) are not considered?
> * _Ressources_:
>   
> - [x]  I suggest changing the order of resources and placing "GemGIS" first and then adding "GemPy" indicating that it is an optional dependency. --> **The order was changed, see https://github.com/cgre-aachen/gemgis_data/commit/32e6fcafb79d17f9801aa5f07730be4f65e3930a**
> - [x]  "Tutorial repository" link and "GemGIS link" are redundant. Choose one or another or change the first to "https://gemgis.readthedocs.io/en/latest/getting_started/tutorial/index.html" --> **The links were combined, see https://github.com/cgre-aachen/gemgis_data/commit/32e6fcafb79d17f9801aa5f07730be4f65e3930a**
> - [x]  _References_:  it looks like the references were broken during the paper compilation. --> **The references compile perfectly using Github Actions in the main repository, see https://github.com/cgre-aachen/gemgis_data/actions/runs/3918599262**

**Comment by acocac** [comment]
> Hi @AlexanderJuestel! I'm on vacation break until January 10th. I'll get back to you asap. Cheers,

@AlexanderJuestel are you still waiting any feedback from @BenjMy?

**Comment by BenjMy** [comment]
Hi @acocac ! From my point of view, I gave all the feedback needed. But happy to contribute to new ones if necessary!

**Comment by acocac** [comment]
> Hi @acocac ! From my point of view, I gave all the feedback needed. But happy to contribute to new ones if necessary!

@BenjMy thanks for confirming you provided all the feedback needed.

**Comment by AlexanderJuestel** [comment]
Dear @acocac, dear @BenjMy,

thanks for checking in! I am making progress and the comments are very good! As you can see from the link above (https://github.com/openjournals/jose-reviews/issues/185#issuecomment-1382733498), I am still working on some points here that may take some more or less time. I will try to work on them in a timely manner but my PhD is keeping my quite busy these days ;) 

I also have to rework the notebooks still but maybe we can check off some open points once I am done with the none-notebook related issues, also to see what is left :) 

Cheers Alex

**Comment by AlexanderJuestel** [comment]
Dear @acocac,

I would like to ask @jwagemann and @BenjMy to go through their comments and remarks and check what I have already completed. In addition, I am sure that some of the tasks at the top of the issue can be checked off already.  

It would be amazing to then have an updated list of what I still have to tackle as things are going all over the place by now 

Cheers 
Alex

**Comment by acocac** [comment]
Hi @jwagemann and @BenjMy. Following @AlexanderJuestel update, I kindly invite you to revisit the checklists and confirm if all suggestions have been addressed in the resource and/or paper.

**Comment by AlexanderJuestel** [comment]
> Hi @jwagemann and @BenjMy. Following @AlexanderJuestel update, I kindly invite you to revisit the checklists and confirm if all suggestions have been addressed in the resource and/or paper.

There will be open points remaining for sure but looking forward to completing some of the other points already :)

**Comment by BenjMy** [comment]
Hi @acocac and @AlexanderJuestel ,

Thanks for the update. I will go over the items and tick the completed ones next week. 
Looking forward to seeing the progress.
Cheers,

Ben

**Comment by BenjMy** [comment]
Hi again @acocac  and @AlexanderJuestel ,

I went through the items and validated almost everything except "Pedagogy" and "Instructional Design" where still some efforts are necessary. I guess it is in a good way, I've seen a lot of improvement already for the documentation and readme.

I'll be happy to read again the paper entirely and check the flow of the documentation/notebooks once https://github.com/cgre-aachen/gemgis_data/issues/18 will be tackled.

Well done!

**Comment by acocac** [comment]
> Hi again @acocac and @AlexanderJuestel ,
> 
> I went through the items and validated almost everything except "Pedagogy" and "Instructional Design" where still some efforts are necessary. I guess it is in a good way, I've seen a lot of improvement already for the documentation and readme.
> 
> I'll be happy to read again the paper entirely and check the flow of the documentation/notebooks once [cgre-aachen/gemgis_data#18](https://github.com/cgre-aachen/gemgis_data/issues/18) will be tackled.
> 
> Well done!

Thanks @BenjMy for all your efforts. @AlexanderJuestel please notify both reviewers when you address the suggested issue.

**Comment by AlexanderJuestel** [comment]
I am making good progress on reworking the tutorial notebooks. I hope to be done with it in one to two weeks if my work schedule allows for that!

![image](https://user-images.githubusercontent.com/45469915/221350732-a623e188-44fa-48e9-b586-d44f25508918.png)
![image](https://user-images.githubusercontent.com/45469915/221350756-487849f1-a929-4a91-97b6-ff25e4dbd846.png)

**Comment by AlexanderJuestel** [comment]
@acocac, @BenjMy, @jwagemann I have just completed the reworking of the notebooks necessary for the tutorial series. I will revise the manuscript a little but everything is ready for you to review it again I would say. I hope I could meet your expectations but are happy to add some more content if needed

Cheers 
Alex

**Comment by acocac** [comment]
> @acocac, @BenjMy, @jwagemann I have just completed the reworking of the notebooks necessary for the tutorial series. I will revise the manuscript a little but everything is ready for you to review it again I would say. I hope I could meet your expectations but are happy to add some more content if needed
> 
> Cheers Alex

Thanks for the update @AlexanderJuestel. I kindly ask @BenjMy @jwagemann to revisit their checklists.

**Comment by acocac** [comment]
@jwagemann I'm wondered if you have any updates of your revision. It would be great to then have an updated list accordingly. Thank you!

**Comment by BenjMy** [comment]
Dear @AlexanderJuestel and Gemgis team,

Congrats again on your hard work and dedication in creating this valuable package. I am certain that it will be enlightening for many users. However, I do have a few minor issues and comments that I would like to bring to your attention.

- Could you consider improving "_The organization of the notebooks does not allow an easy understanding of what notebooks belong to what unit as described in the paper_". 
I see you renamed the notebooks but could you somehow guide the reader with sections and subsections? This join https://github.com/cgre-aachen/gemgis_data/issues/18 of @jwagemann's issue that for me is not yet completely tackled.
This could be (just as an indicative example but up to you to appreciate better how to name the section):
  - Data extraction methods
  - Meshing tools
  - Processing from Pyvista

- Add the reference to the data source for each notebooks in "tutorial and basics usage" (just like you did for the "example model" section)
- Notebooks 59 to 61 are empty
- Missing logo on the documentation page

- in the documentation, section installation: I would rephrase "Installing GemGIS and its dependencies manually" by "Installing GemGIS's dependencies manually" as GEMGIS has already been installed the line before

- api_reference section is empty in the documentation

Let me know if you want me to create issues on the repo directly. 
Best,

Benjamin

**Comment by acocac** [comment]
According to a recent e-mail communitcation with @jwagemann, she will have a look at her checklisr after April 18. Thanks for your hard work too!

**Comment by AlexanderJuestel** [comment]
@BenjMy thanks for your comments! No need to create issues I think :) 

I will fix the documentation, it is actually broken currently due to reasons I still have to figure out....
- Logo: the link is provided in the conf.py file and the file is also provided but I have no idea why it is not showing up. The favicon does though
- The latest notebooks have been added to the main repository and will show up once the documentation compiles again. 
- I will fix the installation documentation
- I tried adding an API reference but failed so far. I had one once before but sphinx is playing some games here
- I can add the references to the tutorial notebooks even though they are actually not a part of the course but good hint :)

Regarding the organization of the Notebooks, I added a table of contents in the introduction notebook and added folders for each section mentioned in the introduction notebook. Within several of the notebooks, I tried to include some post-processing tasks rather than having a dedicated post-processing section.

![image](https://user-images.githubusercontent.com/45469915/229116967-11a91c09-5c92-4e32-bcc6-2016ee05c2f6.png)
![image](https://user-images.githubusercontent.com/45469915/229117132-3070bddf-eab6-4809-b44c-8d6f291fa7cb.png)

**Comment by AlexanderJuestel** [comment]
I have opened an issue with respect to the failing compilation of the documentation here: https://github.com/executablebooks/sphinx-book-theme/issues/719

Waiting on a fix now :)

**Comment by acocac** [comment]
@jwagemann I'd appreciate to confirm if you have any estimated date to complete the checklist as discussed.

**Comment by AlexanderJuestel** [comment]
> I have opened an issue with respect to the failing compilation of the documentation here: [executablebooks/sphinx-book-theme#719](https://github.com/executablebooks/sphinx-book-theme/issues/719)
> 
> Waiting on a fix now :)

The docs are working again

**Comment by acocac** [comment]
> > I have opened an issue with respect to the failing compilation of the documentation here: [executablebooks/sphinx-book-theme#719](https://github.com/executablebooks/sphinx-book-theme/issues/719)
> > Waiting on a fix now :)
> 
> The docs are working again

@BenjMy may I ask you to revisit your review checklist and comments according to the latest update from @AlexanderJuestel?

**Comment by acocac** [comment]
Reminder set for @BenjMy @jwagemann in two days

**Comment by acocac** [comment]
@whedon remind @BenjMy @jwagemann in two days

**Comment by whedon** [comment]
Reminder set for @BenjMy @jwagemann in two days

**Comment by BenjMy** [comment]
Hello @acocac and @AlexanderJuestel ,

Thanks to the GemGIS team for their wonderful work. As you can see I ticked all my checklist and all my issues were tackled. I'm already a happy user of GemGIS and hope for the best for its future dev. I have no further comment and I'm looking forward to seeing soon the JOSE paper online.

**Comment by acocac** [comment]
> Hello @acocac and @AlexanderJuestel ,
> 
> Thanks to the GemGIS team for their wonderful work. As you can see I ticked all my checklist and all my issues were tackled. I'm already a happy user of GemGIS and hope for the best for its future dev. I have no further comment and I'm looking forward to seeing soon the JOSE paper online.

Amazing. Thanks for all your efforts on this. We hope to complete the review process according to the pending revision by the another reviewer.

**Comment by acocac** [comment]
@whedon remind @jwagemann in one day

**Comment by whedon** [comment]
Reminder set for @jwagemann in one day

**Comment by whedon** [comment]
:wave: @jwagemann, please update us on how your review is going (this is an automated reminder).

**Comment by jwagemann** [comment]
Hi. I am just about to start the review. Has the paper been updated? I only found the paper generated on 1 December 2022?

**Comment by jwagemann** [comment]
Hi @AlexanderJuestel ,
thank you for the great revisions to the tutorial. It is now much clearer and also easier to follow for a reader / learner.

Before I can tick all boxes, I have some small suggestions for improvement:
* Version: I still see a version mismatch - last release of the repo was on 29 December 2022 with v.0.0.1?
* README needs to be updated:
  * Target audience: there is a spelling mistake ‘Lectures’ and it should read ‘Lecturers’
  * List of dependencies on README: 
       * Would be good to list the packages together with their versions used
  * The outline seems to be slightly different from the outline [here](https://github.com/cgre-aachen/gemgis_data/blob/main/notebooks/00_introduction_to_structural_modeling.ipynb), correct?
* Links on the outline needs to be updated:
  * Try to make it ‘browsable’ via nbviewer: https://nbviewer.org/github/cgre-aachen/gemgis_data/blob/main/notebooks/00_introduction_to_structural_modeling.ipynb#installation-of-gempy-and-gemgis
  * You have to use relative paths for the notebook links: If I want to open a notebook under 2 or 3, I get referred back to Github
  
I did not revise the paper, as it seems that it has not been updated, correct? The only version I found was from 1 December 2022.

**Comment by AlexanderJuestel** [comment]
@jwagemann thanks for checking some boxes! Let me get back to your comments right away!

- I would suggest a bumping of the version and a new release once the paper is ready for publication. Would that be okay with you?

- The paper was updated a couple of times. I have attached the latest build of the paper from 3 weeks ago (https://github.com/cgre-aachen/gemgis_data/actions/runs/4689904821) 

- The spelling mistake was fixed in https://github.com/cgre-aachen/gemgis_data/commit/0256472d714cc8feff01da62a74b4add4890d21e

- The main dependencies and their tested versions were added in https://github.com/cgre-aachen/gemgis_data/commit/5b635a267102045174fb2ab788fa7205e66c84e0

- I adapted the outline and added NBViewer links to the notebooks linked in the readme in https://github.com/cgre-aachen/gemgis_data/commit/cc91fd46b19c9972937f15553114b82947555547

I will work on the relative paths tomorrow and post an update as soon as possible :) 

[paper.zip](https://github.com/openjournals/jose-reviews/files/11413063/paper.zip)

**Comment by jwagemann** [comment]
Hi @AlexanderJuestel ,
thanks for you swift response. I think you have to activate the command `@whedon generate pdf` to regenerate your pdf in the JOSE format. I had a look at the md file and I ticked the remaining boxes.

@acocac All good on my side, after Alex changed the paths of the notebooks. I can't tick the box with the version, but I am happy as Alex proposed, to change the version to 1.0.0 once the paper will be published.

**Comment by AlexanderJuestel** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

```

**Comment by AlexanderJuestel** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00185/jose.00185/10.21105.jose.00185.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00185/jose.00185/10.21105.jose.00185.pdf) :page_facing_up: :point_left:

**Comment by AlexanderJuestel** [comment]
Good morning @acocac and @jwagemann, 

I changed the links according to the recommendations of @jwagemann in https://github.com/cgre-aachen/gemgis_data/commit/2c4666f2b68bbaa80cba0f75795cc693d3ff9582. To be honest, I did not know NBViewer before and this adds a lot of value to everything! Thanks for the hint! 

If you are fine with everything now, I would create a 1.0.0 release and the submission would be ready for publication from my side :)

**Comment by AlexanderJuestel** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.5281/zenodo.4569086 is OK
- 10.5194/gmd-12-1-2019 is OK
- 10.21105/joss.01450 is OK
- 10.5281/zenodo.4572994 is OK
- 10.1109/MCSE.2007.55 is OK
- 10.1016/j.pepi.2008.06.013 is OK
- 10.1007/978-94-011-2556-7_11 is OK
- 10.1130/GES02455.1 is OK
- 10.1007/BF02775087 is OK
- 10.1007/978-1-4615-9630-1 is OK
- 10.21105/joss.03709 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by acocac** [comment]
> If you are fine with everything now, I would create a 1.0.0 release and the submission would be ready for publication from my side :)

@jwagemann, only a last item in your checklist, can you confirm if everything is ok with the version? thank you

**Comment by AlexanderJuestel** [comment]
If there are no further comments, I will release the final version so @jwagemann  can check her last box

**Comment by AlexanderJuestel** [comment]
I have created a 1.0.0 release: https://github.com/cgre-aachen/gemgis_data/releases/tag/v1.0.0

The Zenodo integration failed right now. I am working on a fix!

![image](https://user-images.githubusercontent.com/45469915/237060289-f365cdbb-45fa-4a14-bf8b-0dd97ada2969.png)

**Comment by AlexanderJuestel** [comment]
@acocac, do you think we can proceed with the publication of the article these days?

Cheers
Alex

**Comment by acocac** [comment]
> @acocac, do you think we can proceed with the publication of the article these days?
> 
> Cheers Alex

Hi @AlexanderJuestel, I'm trying to follow the editorial procedure of JOSE. Please @jwagemann confirm all is ok with the version and complete the checklist 🙏

**Comment by AlexanderJuestel** [comment]
@acocac it seems like @jwagemann has checked all the boxes :)

**Comment by acocac** [comment]
> @acocac it seems like @jwagemann has checked all the boxes :)

Hi @AlexanderJuestel glad to hear the checklist is complete!  Congrats for the hardwork 🚀 

Let me proceed with the publication between today evening and tomorrow. Thanks for your understanding.

**Comment by whedon** [comment]
Wordcount for `paper.md` is 2694

**Comment by acocac** [comment]
> I have created a 1.0.0 release: https://github.com/cgre-aachen/gemgis_data/releases/tag/v1.0.0
> 
> The Zenodo integration failed right now. I am working on a fix!
> 
> ![image](https://user-images.githubusercontent.com/45469915/237060289-f365cdbb-45fa-4a14-bf8b-0dd97ada2969.png)

@AlexanderJuestel Any updates on the issue for archiving v1.0.0 in Zenodo?

**Comment by AlexanderJuestel** [comment]
@acocac, I uploaded the repository manually. You can find it at: [10.5281/zenodo.7938495](https://doi.org/10.5281/zenodo.7938495)

**Comment by acocac** [comment]
@whedon set 10.5281/zenodo.7938495 as archive

**Comment by acocac** [comment]
@whedon set 1.0.0 as version

**Comment by whedon** [comment]
OK. 1.0.0 is the version.

**Comment by acocac** [comment]
@whedon generate pdf

**Comment by acocac** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.5281/zenodo.4569086 is OK
- 10.5194/gmd-12-1-2019 is OK
- 10.21105/joss.01450 is OK
- 10.5281/zenodo.4572994 is OK
- 10.1109/MCSE.2007.55 is OK
- 10.1016/j.pepi.2008.06.013 is OK
- 10.1007/978-94-011-2556-7_11 is OK
- 10.1130/GES02455.1 is OK
- 10.1007/BF02775087 is OK
- 10.1007/978-1-4615-9630-1 is OK
- 10.21105/joss.03709 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by acocac** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.5281/zenodo.4569086 is OK
- 10.5194/gmd-12-1-2019 is OK
- 10.21105/joss.01450 is OK
- 10.5281/zenodo.4572994 is OK
- 10.1109/MCSE.2007.55 is OK
- 10.1016/j.pepi.2008.06.013 is OK
- 10.1007/978-94-011-2556-7_11 is OK
- 10.1130/GES02455.1 is OK
- 10.1007/BF02775087 is OK
- 10.1007/978-1-4615-9630-1 is OK
- 10.21105/joss.03709 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by acocac** [comment]
Hi @labarba, this paper is ready to be published.

@BenjMy @jwagemann, thanks for volunteering your time to review this submission!

@AlexanderJuestel and collaborators, congrats on having this published 🚀

**Comment by acocac** [comment]
@AlexanderJuestel this is just to let you know @labarba who leads the rest of the publication according to her JOSE EiC role got sick recently. Apologies for inconvenience. Thanks for your understanding.

**Comment by AlexanderJuestel** [comment]
@acocac, any update on when @labarba may finish the review?

Still best wishes to her in the meantime!
Cheers
Alex

**Comment by labarba** [comment]
@AlexanderJuestel Thank you for your patience. I'm coming out of COVID with already a week-long backlog. I'm also editor-in-chief of another journal. My response times are longer than two weeks.

**Comment by acocac** [comment]
Dear @AlexanderJuestel, fyi, Lorena confirmed me today there's still a backlog on publishing the final papers. We appreciate your patience.

**Comment by AlexanderJuestel** [comment]
Hello @acocac, just wondering if there were any news since it has been another four weeks already.

Cheers
Alex

**Comment by acocac** [comment]
> Hello @acocac, just wondering if there were any news since it has been another four weeks already.
> 
> Cheers Alex

@labarba we wondered if you have updates to proceed with the publication of this submission 🙏

**Comment by labarba** [comment]
Hi everybody! 👋 
Thank you all for your patience. I looked over the paper, and see that the citations need fixing. 
@AlexanderJuestel – Please have a look a the [Markdown citation format](https://pandoc.org/MANUAL.html#citations) we use.

**Comment by labarba** [comment]
From our [docs](https://openjournals.readthedocs.io/en/jose/submitting.html#example-paper-and-bibliography):

# Citations

Citations to entries in paper.bib should be in
[rMarkdown](http://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html)
format.

For a quick reference, the following citation commands can be used:
- `@author:2001`  ->  "Author et al. (2001)"
- `[@author:2001]` -> "(Author et al., 2001)"
- `[@author1:2001; @author2:2001]` -> "(Author1 et al., 2001; Author2 et al., 2002)"

**Comment by AlexanderJuestel** [comment]
@whedon generate pdf

**Comment by AlexanderJuestel** [comment]
@editorialbot generate pdf

**Comment by AlexanderJuestel** [comment]
@labarba I have fixed the citation commands in the manuscript and also adapted the rules for capitalization in titles in the bibtex file.

Let me know if there is anything else. 

Cheers Alex

**Comment by labarba** [comment]
Ah, yes. Some links are broken too. Have a look at the bottom of page 7.

**Comment by AlexanderJuestel** [comment]
Thanks for pointing that out, @labarba. The links should be fixed now!

**Comment by labarba** [comment]
@editorialbot generate pdf

**Comment by labarba** [comment]
We have another error. Please check the report above. It says:

`Error reading bibliography file paper.bib:
(line 175, column 1):
unexpected '@'`

**Comment by AlexanderJuestel** [comment]
Yeah, forgot to merge my PR, sorry

**Comment by AlexanderJuestel** [comment]
@editorialbot generate pdf

**Comment by labarba** [comment]
This is such an impressive piece of work!

I could publish as is, but maybe you want to correct a couple of remaining details: on p.3, line 119, the phrase "for the basic structural geological," seems to be missing a noun that follows, like "model" or "map" or something like that - on p.6, line 165, "the open-source developed 3D creation suite Blender" sounds awkward; maybe just delete "developed" - in the Bibliography, use brackets to protect the author display that now awkwardly shows "Community, B. O"

**Comment by AlexanderJuestel** [comment]
Thank you! The comments of both reviewers improved the quality of the entire submission. 

I fixed your remaining comments!

**Comment by labarba** [comment]
@editorialbot generate pdf

**Comment by labarba** [comment]
@editorialbot accept

**Comment by labarba** [comment]
Congratulations @AlexanderJuestel, your JOSE paper is published! 🚀 

Huge thanks to our Editor: @acocac and Reviewers: @BenjMy, @jwagemann — your contributions make this possible 🙏

**Comment by AlexanderJuestel** [comment]
@labarba great news! Thank you very much! Thank you to both reviewers (@jwagemann, @BenjMy) and to @acocac for managing everything!

**Comment by labarba** [comment]
:loudspeaker:  New Reviewer Sign-up Form!
https://forms.gle/Rizd3TcHnQKhrbYY7
